### PR TITLE
Document new authentication features

### LINKED ISSUE #152: Document new authentication features
Introduced by prometheus/prometheus#921

### PR DESCRIPTION
**Pull request by jimmidyson**
Fixes #152

### CODE REVIEW

File: content/docs/operating/configuration.md [hunk 1]
```diff
@@ -83,6 +83,15 @@ dynamically discovered using one of the supported service-discovery mechanisms.
 Additionally, `relabel_configs` allow advanced modifications to any
 target and its labels before scraping.
 
+If the target requires authentication, the following options are available:
```
Comment (fabxc) [inline content/docs/operating/configuration.md]: target**s**

File: content/docs/operating/configuration.md [hunk 2]
```diff
@@ -83,6 +83,15 @@ dynamically discovered using one of the supported service-discovery mechanisms.
...
+
+* `basic_auth`
```
Comment (fabxc) [inline content/docs/operating/configuration.md]: I know this list will be self-explaining to many, but maybe a brief description after each point?

**Comment by fabxc** [comment]
Thanks! :+1: from my side.

@juliusv might want to check as well.

File: content/docs/operating/configuration.md [hunk 3]
```diff
@@ -83,6 +83,20 @@ dynamically discovered using one of the supported service-discovery mechanisms.
 Additionally, `relabel_configs` allow advanced modifications to any
 target and its labels before scraping.
 
+If the targets require authentication, the following options are available:
+
+* `basic_auth` - sets the `Authorization` header on every scrape request with the
+configured username & password.
```
Comment (juliusv) [inline content/docs/operating/configuration.md]: s/&/and/ here and below please.

File: content/docs/operating/configuration.md [hunk 4]
```diff
@@ -114,11 +128,25 @@ job_name: <name>
 # The URL scheme with which to fetch metrics from targets.
 [ scheme: <scheme> | default = http ]
 
-# HTTP basic authentication information.
+# Optional authentication information. Note that basic_auth, bearer_token &
```
Comment (juliusv) [inline content/docs/operating/configuration.md]: Dito here.

Comment (juliusv) [inline content/docs/operating/configuration.md]: Please put backticks around `basic_auth`, `bearer_token`, ... to format them as code.

Comment (jimmidyson) [inline content/docs/operating/configuration.md]: Done.

File: content/docs/operating/configuration.md (hunk 3 above)
Comment (jimmidyson) [inline content/docs/operating/configuration.md]: Done.

File: content/docs/operating/configuration.md [hunk 5]
```diff
@@ -83,6 +83,20 @@ dynamically discovered using one of the supported service-discovery mechanisms.
...
+configured username & password.
+* `client_cert` - configures the scrape request to use
+[mutual TLS](https://en.wikipedia.org/wiki/Mutual_authentication) with the
+configured certificate & key.
+* `bearer_token` - sets the `Authorization` header on every scrape request with
+the configured bearer token.
+* `bearer_token_file` - sets the `Authorization` header on every scrape request
+with the bearer token read from the configured file.
+
+See below for example use.
```
Comment (juliusv) [inline content/docs/operating/configuration.md]: It's not really an example use, but more of a reference of where those options go, right? How about "See below for the configuration of these authentication options.".

**Comment by juliusv** [comment]
:+1: otherwise

File: content/docs/operating/configuration.md (hunk 5 above)
Comment (jimmidyson) [inline content/docs/operating/configuration.md]: Done.

**Comment by juliusv** [comment]
Thanks! :)

**Pull request merged**